Release v2.7.0 (#6854)
aiidateam/aiida-core: CHANGELOG.md

```python
node = Int(5) # Can be any ORM node
serialized_node = node.serialize()
print(serialized_node)
# {'pk': None, 'uuid': '485c2ec8-441d-484d-b7d9-374a3cdd98ae', 'node_type': 'data.core.int.Int.', 'process_type': None, 'repository_metadata': {}, 'ctime': datetime.datetime(2025, 5, 2, 10, 20, 41, 275443, tzinfo=datetime.timezone(datetime.timedelta(seconds=7200), 'CEST')), 'mtime': None, 'label': '', 'description': '', 'attributes': {'value': 5}, 'extras': {}, 'computer': None, 'user': 1, 'repository_content': {}, 'source': None, 'value': 5}
Warning: Serialization through pydantic is still an experimental feature and might break in future releases.
uuid: 77e9c19a-5ecb-40cf-8238-ea5c55fbb83f (unstored) value: 5
node_deserialized = Int.from_serialized(**serialized_node)
print(node_deserialized)
# uuid: 77e9c19a-5ecb-40cf-8238-ea5c55fbb83f (unstored) value: 5
```

Note that this is still not a stable feature and might break in future releases without deprecation warning. For more information see [AEP 010](https://github.com/aiidateam/AEP/blob/983a645c9285ba65c7cf07fe6064c23e7e994c06/010_orm_schema/readme.md)

#### Stashing (PR #6844, #6763)

> TODO

Reviewer: HERE

> #### Stashing [#6844]((https://github.com/aiidateam/aiida-core/pull/6844)), [#6763](https://github.com/aiidateam/aiida-core/pull/6763)
> 
> TODO

Reviewer: ```suggestion
#### Stashing [#6772](https://github.com/aiidateam/aiida-core/pull/6772), [#6746](https://github.com/aiidateam/aiida-core/pull/6746)

Historically, stashing was only possible, if it was instructed before running a generic calcjob. The instruction had to be "attached" to the original calcjob, like this for example:

```python
inputs = {
    'MyInputs': <MyInputs>,
    'metadata': {
        'computer': Computer.collection.get(label="localhost"),
        'options': {
            'resources': {'num_machines': 1}, 
            'stash': {
                'stash_mode': StashMode.COPY.value,
                'target_base': '/scratch/',
                 'source_list': ['heavy_data.xyz'],
            },
        },
    },
}
run(MyCalculation, **inputs)```

However, if a user would realize they need to stash something only after running a calcjob, this would not be possible.

Now, we introduced a new calcjob, which is able to perform a stashing operation after a calculation is finished.
The usage is very similar, and for consistency and user-friendliness, we keep the instruction as part of the metadata. The only main input is obviously a source node which is `RemoteData` node of the calculation to be stashed, for example:

```python
StashCalculation_ = CalculationFactory('core.stash')

MyCalculation = orm.load_node(pk=<PK>)
inputs = {
    'metadata': {
        'computer': Computer.collection.get(label="localhost"),
        'options': {
            'resources': {'num_machines': 1}, 
            'stash': {
                'stash_mode': StashMode.COPY.value,
                'target_base': '/scratch/',
                 'source_list': ['heavy_data.xyz'],
            },
        },
    },
    'source_node': orm.RemoteData,
}

result = run(StashCalculation_, **inputs)```

In addition, we now support more types for `stash_mode`, allowing for compressed operations:
'stash_mode': StashMode.COPY.value,
'stash_mode': StashMode.COMPRESS_TAR
'stash_mode': StashMode.COMPRESS_TARBZ2
'stash_mode': StashMode.COMPRESS_TARGZ
'stash_mode': StashMode.COMPRESS_TARXZ

```

    ...,
    'metadata': {
        'computer': Computer.collection.get(label="localhost"),
        'options': {
            'resources': {'num_machines': 1}, 
            'stash': {
                'stash_mode':  "tar.gz",
                'target_base': '/scratch/',
                 'source_list': ['heavy_data.xyz'],
            },
        },
    },
}
run(MyCalculation, **inputs)
```

Historically, stashing was only possible, if it was instructed before running a generic calcjob. The instruction had to
be "attached" to the original calcjob. However, if a user would realize they need to stash something only after running a calcjob, this would not be possible.

Now, we introduced a new calcjob `StashCalculation` which is able to perform a stashing operation after a calculation has finished. The usage is very similar, and for consistency and user-friendliness, we keep the instruction as part of the metadata. The only main input is obviously a source node which is the UUID of the `RemoteData` node of the calculation to be stashed, for example:

```python
from aiida.orm import load_node

StashCalculation = CalculationFactory('core.stash')

calcjob_node = load_node(<CALCJOB_PK>)

> remote_folder = cj.outputs.remote_folder

Author: Changelog has been overhauled

These enhancements aim to make data export from AiiDA more robust, customizable, and user-friendly.

### Stashing ([#6746](https://github.com/aiidateam/aiida-core/pull/6746), [#6772](https://github.com/aiidateam/aiida-core/pull/6772))

With this feature, you can bundle your data to a (compressed) tar archive during stashing by specifying one of the `stash_mode` options `"tar"`, `"tar.bz2"`, `"tar.gz"`, or `"tar.xz"`.
When specifying the stashing operation during the setup of your calculation, compression can be configured as follows:

```python
from aiida.plugins import CalculationFactory
from aiida.engine import run
from aiida.common import StashMode
from aiida.orm import load_computer

inputs = {
    ...,
    'metadata': {
        'computer': load_computer(label="localhost"),
        'options': {
            'resources': {'num_machines': 1},
            'stash': {

>                 'stash_mode':  StashMode.COMPRESS_TARGZ,

Author: ```suggestion
                'stash_mode':  StashMode.COMPRESS_TARGZ.value,
```

>                 'stash_mode':  StashMode.COMPRESS_TARGZ,

Reviewer: `StashMode.COMPRESS_TARGZ.value`
```suggestion
                'stash_mode':  StashMode.COMPRESS_TARGZ.value,
```

Note that this is still not a stable feature and might break in future releases without deprecation warning. For more information see [AEP 010](https://github.com/aiidateam/AEP/blob/983a645c9285ba65c7cf07fe6064c23e7e994c06/010_orm_schema/readme.md)

#### Stashing [#6746](https://github.com/aiidateam/aiida-core/pull/6746), [#6772](https://github.com/aiidateam/aiida-core/pull/6772)

You can now bundle your data to a tar during the stashing (with compression) by specifying one of the `stash_mode`s option `"tar"`, `"tar.bz2"`, `"tar.gz"`, `"tar.xz"`.

```python
inputs = {
    ...,
    'metadata': {
        'computer': Computer.collection.get(label="localhost"),
        'options': {
            'resources': {'num_machines': 1}, 
            'stash': {

>                 'stash_mode':  "tar.gz",

Reviewer: Not recommended. 
This goes against the logic of Enum data class. 
If we change `"tar.gz"` to `"targz"` will break every thing..
